expertsystem/topology.py

```diff
@@ -152,11 +154,40 @@ def get_originating_node_list(
             self.edges[edge_id].originating_node_id for edge_id in edge_ids
         ]
 
-    def verify(self) -> bool:  # pylint: disable=no-self-use
-        """Verify if the graph is connected.
+    def verify(self) -> None:
+        """Verify if there are no dangling edges or nodes."""
+        node_ids = self.nodes | {None}  # trick to ignore None
+        for edge_id, edge in self.edges.items():
+            if (
+                edge.originating_node_id is None
+                and edge.ending_node_id is None
+            ):
```

Comment:
You could call `get_connected_nodes()` here

Reply:
3541b385 and 8cee9310